src/transactions/transactionBytes.js

```diff
@@ -12,323 +12,252 @@
-	 * @param transactionBuffer buffer
-	 * @param assetSize number
-	 * @param assetBytes number
-	 * @return {buffer}
-	 */
-
-	function assignTransactionBuffer(transactionBuffer, assetSize, assetBytes) {
-		transactionBuffer.writeInt8(transaction.type);
-		transactionBuffer.writeInt(transaction.timestamp);
-
-		assignHexToTransactionBytes(transactionBuffer, transaction.senderPublicKey);
-
-		if (transaction.requesterPublicKey) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.requesterPublicKey);
-		}
-
-		if (transaction.recipientId) {
-			let recipient = transaction.recipientId.slice(0, -1);
-			recipient = bignum(recipient).toBuffer({ size: 8 });
-
-			for (let i = 0; i < 8; i++) {
-				transactionBuffer.writeByte(recipient[i] || 0);
-			}
-		} else {
-			for (let i = 0; i < 8; i++) {
-				transactionBuffer.writeByte(0);
-			}
-		}
-		transactionBuffer.writeLong(transaction.amount);
-
-		if (transaction.asset.data) {
-			const dataBuffer = Buffer.from(transaction.asset.data);
-			for (let i = 0; i < dataBuffer.length; i++) {
-				transactionBuffer.writeByte(dataBuffer[i]);
-			}
-		}
-
-		if (assetSize > 0) {
-			for (let i = 0; i < assetSize; i++) {
-				transactionBuffer.writeByte(assetBytes[i]);
-			}
-		}
-
-		if (transaction.signature) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.signature);
-		}
-
-		if (transaction.signSignature) {
-			assignHexToTransactionBytes(transactionBuffer, transaction.signSignature);
-		}
-
-		transactionBuffer.flip();
-		const arrayBuffer = new Uint8Array(transactionBuffer.toArrayBuffer());
-		const buffer = [];
-
-		for (let i = 0; i < arrayBuffer.length; i++) {
-			buffer[i] = arrayBuffer[i];
+export function checkTransaction(transaction) {
+	if (transaction.asset.data) {
+		if (transaction.asset.data.length > BYTESIZES.DATA) {
+			throw new Error(`Transaction asset data exceeds size of ${BYTESIZES.DATA}.`);
 		}
-
-		return Buffer.from(buffer);
 	}
-
-	// Get Transaction Size and Bytes
-	const transactionAssetSizeBuffer = getEmptyBytesForTransaction(transaction);
-	const assetSize = transactionAssetSizeBuffer.assetSize;
-	const assetBytes = transactionAssetSizeBuffer.assetBytes;
-
-	const emptyTransactionBuffer = createEmptyTransactionBuffer(assetSize);
-	const assignedTransactionBuffer = assignTransactionBuffer(
-		emptyTransactionBuffer, assetSize, assetBytes,
-	);
-
-	return assignedTransactionBuffer;
 }
 
-/**
- * @method getBytes
- * @param transaction Object
- *
- * @return {buffer}
- */
+const REQUIRED_TRANSACTION_PARAMETER = [
+	'type',
+	'timestamp',
+	'senderPublicKey',
+	'amount',
+];
 
-function getTransactionBytes(transaction) {
-	return createTransactionBuffer(transaction);
+/**
+* A utility class to get transaction byteSizes
+*
+* @class TransactionBytes
+* @param {Object} transaction
+* @constructor
+*/
+
+export function getTransactionBytes(transaction) {
+	checkTransaction(transaction);
+	checkRequiredFields(REQUIRED_TRANSACTION_PARAMETER, transaction);
```

Comment:
Maybe this should be called inside `checkTransaction` since that sounds like it's going to check everything.